src/ssr-template-helper.js

```diff
@@ -96,17 +97,21 @@ export class SsrTemplateHelper {
 
   /**
    * @param {!Element} element
-   * @param {!JsonObject} data
+   * @param {!Array|!JsonObject|string} data
    * @return {!Promise}
    */
   renderTemplate(element, data) {
     let renderTemplatePromise;
-    if (this.isSupported()) {
+    if (this.isSupported()
+        && data['html'] && typeof data['html'] === 'string') {
```

Comment:
I think we should split these conditions. What if `isSupported() == true` but `data['html'] == null`?

Reply:
I mean in that case we shouldn't go to the `else` block right?

Reply:
Ah gotcha. Thanks for clearing that up.

Reply:
Hey Will, now I see what you meant. Actually, if SSR is not supported then it should go to the else block as the other clauses are for the cases that is non SSR related. With that in mind, it is now confusing as the class is now delegating to methods that are not SSR related. I think this change is good but perhaps renaming this class template helper and naming the methods related to ssr accordingly would be the desired change. Let me know what you think.

Reply:
I mean if SSR is supported but the data returned is corrupted (doesn't contain `html` key), we should not go to the `else` block. I.e.

```js
if (this.isSupported()) {
  if (typeof data['html'] === 'string') {
  } else {
    // We shouldn't be calling `findAndRenderTemplate` here. Just emit an error and no-op.
  }
} else {
  ...
}
```

Reply:
I still find it confusing though that now SSR template helper is dealing with non SSR things. I think a name change should be considered.

Reply:
Sounds good. Let's keep as is for now. Thanks